src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,184 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.item = item;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
+    private static final int SIZE_EMPTY_LIST = 0;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (size() == SIZE_EMPTY_LIST) {
+            head = newNode;
+        }
+        if (size > SIZE_EMPTY_LIST) {
+            newNode.prev = tail;
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size() || index < SIZE_EMPTY_LIST) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of range.");
+        }
+        if (index == SIZE_EMPTY_LIST) {
+            createHead(value);
+        } else if (index == size()) {
+            createTail(value);
+        } else {
+            createItem(value, index);
+        }
```

Comment:
I think your logic is too complex (in private methods), you can simplify your suggestion without this methods 
```suggestion
        Node<T> currentNode = findByIndex(index);
        Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
        if (index == SIZE_EMPTY_LIST) {
            // simplify this
        } else if (index == size) {
            // you can use add(T value) and you can move this check to before checkIndexOutOfBoundsException(index, size);
        } else {
            // simplify this
        }
```

Reply:
I am not sure
https://github.com/mate-academy/jv-linked-list/pull/609#discussion_r941226192